### PR TITLE
Add support for LUXY Smart Light (ZMNHQD)

### PR DESCRIPTION
**Pull request by TedTolboom**
@RobinBol the pull request to add support for the LUXY Smart Light (ZMNHQD); for code review.
I've build as much as possible based on the existing app structure.

Some minor details (e.g. product images) need to be updated; among others the meshdriver (https://github.com/athombv/node-homey-meshdriver/pull/180) and the device images, but can be done in parallel to the code review.

Looking forward to your feedback.

### CODE REVIEW

File: .homeycompose/drivers/flow/actions/turnOffBuzzer.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "$id": "turnOffBuzzer_{{driverId}}",
+  "title": {
+    "en": "Deactivate alarm buzzer",
```
Comment (RobinBol) [inline .homeycompose/drivers/flow/actions/turnOffBuzzer.json]: Personally, I think 'alarm buzzer' is a bit double. I'd opt for either 'buzzer' (preferably) or 'alarm'. I leave it to you to decide which you think fits best (please make sure if you change it that you change it everywhere).

File: .homeyplugins.json [hunk 2]
```diff
@@ -1,8 +1,5 @@
 [
   {
     "id": "compose"
-  },
-  {
-    "id": "zwave"
```
Comment (RobinBol) [inline .homeyplugins.json]: Please make sure to revert this change before going forward with this PR. The homey-meshdriver PR is in review and will likely be released to production in the coming days.

File: app.json [hunk 3]
```diff
@@ -10256,6 +10312,300 @@
...
+            "en": "Set the alarm buzzer volume",
+            "nl": "Zet het alarm buzzer volume"
+          }
+        },
+        "onoff.buzzer": {
+          "title": {
+            "en": "Alarm buzzer turned on",
+            "nl": "Alarm buzzer aangezet"
+          },
+          "preventInsights": true,
```
Comment (RobinBol) [inline app.json]: Why would you prevent logging this to insights? Besides, it seems you added some insights specific titles, so I guess this is a mistake?

File: app.json [hunk 4]
```diff
@@ -10256,6 +10312,300 @@
...
+          "usageOn": 1.5,
+          "usageOff": 0
+        }
+      },
+      "zwave": {
+        "manufacturerId": 345,
+        "productTypeId": 8,
+        "productId": 81,
+        "includeSecure": false,
+        "__comment": "Reverted back to non secure inclusion due to S2 security handshake issues",
```
Comment (RobinBol) [inline app.json]: Although this currently passes the validator, it is not recommended to add custom JSON properties.

File: app.json [hunk 5]
```diff
@@ -10256,6 +10312,300 @@
...
+        "associationGroups": [
+          1
+        ],
+        "learnmode": {
+          "image": "/drivers/ZMNHQD/assets/learnmode.svg",
+          "instruction": {
+            "en": "(Re-)activate full white (one press on (1)). \n\nWhen full white is enabled, press right (2) and hold it in between 4 and 6 seconds. \n\nAfter 6 seconds, the LUXY Smart Light (ZMNHQD) starts flashing green 1 second on, 0.5 second off.\n\nOnce the device has been succesfully added (after 10 seconds), it stops flashing and turns full green.",
```
Comment (RobinBol) [inline app.json]: Maybe that the learnmode image clarifies these instructions, but for me it is unclear what (1) is. Personally the 'Once the device ... full green.' line is a bit redundant, the UI will indicate when the device is added successfully, lets try to keep these sorts of instructions as to the point as possible:)

File: app.json [hunk 6]
```diff
@@ -10256,6 +10312,300 @@
...
+          "attr": {
+            "min": 0,
+            "max": 32535
+          },
+          "units": {
+            "en": "s",
+            "nl": "s"
+          },
+          "hint": {
+            "en": "This parameter (11) defines the time after which the device is turned off automatically. Note: setting this parameter to zero will disable automatically turning off.",
```
Comment (RobinBol) [inline app.json]: Maybe the hint should also clarify that values between 0 and 30 are not accepted.

File: app.json [hunk 7]
```diff
@@ -10256,6 +10312,300 @@
...
+            "en": "s",
+            "nl": "s"
+          },
+          "value": 0,
+          "attr": {
+            "min": 0,
+            "max": 32535
+          },
+          "hint": {
+            "en": "This parameter (10) defines the time after which the device is turned on automatically. Note: setting this parameter to zero will disable automatically turning on.",
```
Comment (RobinBol) [inline app.json]: See comment for parameter 11.

File: app.json [hunk 8]
```diff
@@ -10256,6 +10312,300 @@
...
+          "label": {
+            "en": "Turn off alarm buzzer",
+            "nl": "Uitschakelen alarm buzzer"
+          },
+          "value": "1",
+          "values": [
+            {
+              "id": "0",
+              "label": {
+                "en": "Only by Z-wave",
```
Comment (RobinBol) [inline app.json]: These options should be clarified, I think here you mean by received commands from other Z-Wave devices (i.e. via associations)?

File: app.json [hunk 9]
```diff
@@ -10256,6 +10312,300 @@
...
+              "id": "1",
+              "label": {
+                "en": "Capacitive input & Z-wave",
+                "nl": "Capacitieve bedieningsknoppen & Z-wave"
+              }
+            }
+          ],
+          "hint": {
+            "en": "This parameter (3) determines how the alarm (buzzer) can be disabled.",
+            "nl": "Deze parameter (3) specificeert hoe het alarm (geluid) uitgeschakeld kan worden."
```
Comment (RobinBol) [inline app.json]: 'geluid' or 'buzzer'? Try to keep these references consistent, makes it easier for users to understand.

File: app.json [hunk 10]
```diff
@@ -10256,6 +10312,300 @@
...
+            "index": 5,
+            "size": 2
+          },
+          "label": {
+            "en": "Motion sensor switch off timer",
+            "nl": "Bewegingssensor uitschakeltijd"
+          },
+          "hint": {
+            "en": "This parameter (5) determines the time after which the output triggered by motion sensor is turned off. \n32535 – never turns off",
+            "nl": "Deze parameter (5) bepaalt de tijdsduur waarna de uitgang die ingeschakeld is doot de Bewegingssensor uitgeschakeld wordt. \n32535 – schakelt nooit uit"
```
Comment (RobinBol) [inline app.json]: Typo 'doot'.

Besides, try not to use '\n' if not really needed. I think I would write this as: '... turned off. Setting the value to 32535 will result in the output never being turned off.' Something like that.

File: drivers/ZMNHQD/assets/icon.svg [hunk 11]
```diff
@@ -0,0 +1,45 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (RobinBol) [inline drivers/ZMNHQD/assets/icon.svg]: Did you request the icons via homey-vectors-public? If not, please do so, this makes sure the icon styling is consistent.

File: drivers/ZMNHQD/device.js [hunk 12]
```diff
@@ -0,0 +1,499 @@
...
+
+const QubinoDimDevice = require('../../lib/QubinoDimDevice');
+const { CAPABILITIES, COMMAND_CLASSES, FLOWS, SETTINGS } = require('../../lib/constants');
+
+const COLOR_REPORT_DEBOUNCE_TIMEOUT = 500; // ms
+const MULTIPLE_CAPABILITIES_DEBOUNCE_TIMEOUT = 500; // ms
+
+/**
+ * LUXY Smart Light (ZMNHQD)
+ * Extended manual: https://qubino.com/manuals/Flush_RGBW_Dimmer.pdf
```
Comment (RobinBol) [inline drivers/ZMNHQD/device.js]: This refers to a different product manual.

File: drivers/ZMNHQD/device.js [hunk 13]
```diff
@@ -0,0 +1,499 @@
...
+ */
+class ZMNHQD extends QubinoDimDevice {
+
+  /**
+   * Method that registers custom setting parsers.
+   */
+  registerSettings() {
```
Comment (RobinBol) [inline drivers/ZMNHQD/device.js]: Unnecessary override.

File: drivers/ZMNHQD/device.js [hunk 14]
```diff
@@ -0,0 +1,499 @@
...
+        return {'Tone identifier': value ? 1: 0,}
+    	},
+    	report: 'SOUND_SWITCH_TONE_PLAY_REPORT',
+    	reportParserV1: report => {
+    		if (report && report.hasOwnProperty('Tone Identifier')) {
+    			if (report['Tone Identifier'] === 1) {
+            this.driver.triggerFlow(FLOWS.BUZZER_TURNED_ON, this, {}, {}).catch(err => this.error('failed to trigger flow', FLOWS.BUZZER_TURNED_ON, err));
+            return true;
+          }
+    			else if (report['Tone Identifier'] === 0) {
```
Comment (RobinBol) [inline drivers/ZMNHQD/device.js]: Seems you have some weird code style issues here and there, it is best to update to our latest eslint config (`npm i --save-dev eslint-config-athom@latest`) and at least fix all auto fixable issues (`eslint --fix ./drivers/ZMNHQD/device.js`), preferably take a look at the remaining non-auto-fixable eslint issues (`eslint ./drivers/ZMNHQD/device.js` will output all warnings and errors).

File: drivers/ZMNHQD/device.js [hunk 15]
```diff
@@ -0,0 +1,499 @@
...
+   * Method that generates a color component object which is then set on the device. This color component object is
+   * generated based on the current and new capability values.
+   * @param {Object} newCapabilitiesValuesObject
+   * @param {Object} currentCapabilitiesValuesObject
+   * @returns {{red: number, green: number, blue: number}}
+   * @private
+   */
+  _generateColorComponents(newCapabilitiesValuesObject, currentCapabilitiesValuesObject) {
+    // Get desired capability values from new/current
+    const onoff = this._getDesiredCapabilityValue('onoff', newCapabilitiesValuesObject, currentCapabilitiesValuesObject);
```
Comment (RobinBol) [inline drivers/ZMNHQD/device.js]: I see you used the `CAPABILITIES` constants in your code, great, however here is still see some strings, maybe you want to change that.

File: drivers/ZMNHQD/device.js [hunk 16]
```diff
@@ -0,0 +1,499 @@
...
+      // Override colorComponent object with stored values
+      colorComponents.red = this._colorComponentsState.red;
+      colorComponents.green = this._colorComponentsState.green;
+      colorComponents.blue = this._colorComponentsState.blue;
+
+      if (typeof this._colorComponentsState.dim === 'number') this.setCapabilityValue(CAPABILITIES.DIM, this._colorComponentsState.dim);
+      else this.setCapabilityValue(CAPABILITIES.DIM, 1);
+
+      // First check if colors need to be updated
+      //this._sendColors({ ...colorComponents });
```
Comment (RobinBol) [inline drivers/ZMNHQD/device.js]: Should this remain commented out?

File: drivers/ZMNHQD/device.js [hunk 17]
```diff
@@ -0,0 +1,499 @@
...
+
+      // Switch on dimmer
+      this.executeCapabilitySetCommand(CAPABILITIES.ONOFF, COMMAND_CLASSES.SWITCH_MULTILEVEL, true, {
```
Comment (RobinBol) [inline drivers/ZMNHQD/device.js]: Should the execution flow stop here? In other words, should we `return this.exectuteCapabilitySetCommand()`?

File: drivers/ZMNHQD/device.js [hunk 18]
```diff
@@ -0,0 +1,499 @@
...
+
+    // Device has been turned off
+    if (red === 0 && green === 0 && blue === 0) {
+
+      // Colors where dimmed to zero so device is basically off, do not continue further
+      return this.setCapabilityValue(CAPABILITIES.ONOFF, false);
+    } if (this.getCapabilityValue(CAPABILITIES.ONOFF) === false && (red > 0 || green > 0 || blue > 0)) {
+
+      // Device was off but received color values that indicate device is on
+      // this.setCapabilityValue(CAPABILITIES.ONOFF, true);
```
Comment (RobinBol) [inline drivers/ZMNHQD/device.js]: Empty `if`?

File: drivers/ZMNHQD/device.js [hunk 19]
```diff
@@ -0,0 +1,499 @@
...
+    red, green, blue, dim,
+  }) {
+    this.log(`_processColorUpdates() -> rgbw(${red},${green},${blue})`);
+
+    // Device has been turned off
+    if (red === 0 && green === 0 && blue === 0) {
+
+      // Colors where dimmed to zero so device is basically off, do not continue further
+      return this.setCapabilityValue(CAPABILITIES.ONOFF, false);
+    } if (this.getCapabilityValue(CAPABILITIES.ONOFF) === false && (red > 0 || green > 0 || blue > 0)) {
```
Comment (RobinBol) [inline drivers/ZMNHQD/device.js]: Should this be an `else if` (based on the indentation)?

File: drivers/ZMNHQD/driver.compose.json [hunk 20]
```diff
@@ -0,0 +1,80 @@
...
+        "nl": "Alarm buzzer geactiveerd"
+      },
+      "insightsTitleFalse": {
+        "en": "Alarm buzzer deactivated",
+        "nl": "Alarm buzzer uitgeschakeld"
+      }
+    }
+  },
+  "energy": {
+		"approximation": {
```
Comment (RobinBol) [inline drivers/ZMNHQD/driver.compose.json]: Indentation (see my eslint comments).

File: lib/QubinoDriver.js [hunk 21]
```diff
@@ -21,6 +21,14 @@ class QubinoDriver extends Homey.Driver {
         key: FLOWS.SET_OFF_AUTO_THERMOSTAT_MODE,
         fn: this.setOffAutoThermostatMode,
       },
+      {
```
Comment (RobinBol) [inline lib/QubinoDriver.js]: Maybe it is better to override onInit() in `ZMNHQD/driver.js`, then after `super.onInit()` call `_registerFlowCardTriggerDevice()` for the Flow cards specific to this driver. The idea was that the generic QubinoDriver should only register the common Flow cards (such as the inputs).

File: lib/QubinoDriver.js [hunk 22]
```diff
@@ -126,6 +136,19 @@ class QubinoDriver extends Homey.Driver {
     throw new Error('missing_device_instance');
   }
 
+  /**
+   * Flow Action handler that will set the buzzer onoff state on a device.
+   * @param value
+   * @param args
+   */
+  async setOnOffBuzzerFlowAction(value, args) {
```
Comment (RobinBol) [inline lib/QubinoDriver.js]: See comment above, I think this should be moved to `ZMNHQD/driver.js`.

File: app.json [hunk 23]
```diff
@@ -10256,6 +10312,300 @@
...
+        },
+        {
+          "type": "group",
+          "label": {
+            "en": "Dimming",
+            "nl": "Dimmen"
+          },
+          "children": [
+            {
+              "id": "dimDuration",
```
Comment (RobinBol) [inline app.json]: Forgot one thing, what is the purpose of this (non-zwave) setting? Maybe I am missing something, but it seems as if it is not used anywhere.

File: app.json (hunk 3 above)
Comment (TedTolboom) [reply app.json]: Well, defining the insight specific titles is not being reflected in the device insights... is this a known issue?
Both onoff and onoff.buzzer show with the same logging, making them confusing; so therefore I disabled Insights (for now)

File: app.json (hunk 4 above)
Comment (TedTolboom) [reply app.json]: I think I copied this one from the Fibaro app... ;-P

File: app.json (hunk 5 above)
Comment (TedTolboom) [reply app.json]: Indeed (1) and (2) are referring to the image shown; positions on the capacitive surface.
The 'Once the device ... full green.' is redundent. 'Once the device ... full red.' is required because of the exclusion / inclusion sequence timing

File: app.json (hunk 23 above)
Comment (TedTolboom) [reply app.json]: It actually is a Z-wave specific, but non-parameter based setting.
It is used to specify the dim duration for any Z-wave command send from Homey to the device; similar to the dim duration option on a action card.

Comment (RobinBol) [reply app.json]: Ah, all right, but where can you link to where the setting used?

File: app.json (hunk 3 above)
Comment (RobinBol) [reply app.json]: Strange, not that I know of. Is the custom capability title applied or also not?

Comment (RobinBol) [reply app.json]: By the way, where do you expect the title to appear? In insights.homey.app or also in the smartphone app (the last device view 'timeline')?

Comment (TedTolboom) [reply app.json]: I would expect on the timeline (which it isn't) and on insights (didn't check).

Comment (RobinBol) [reply app.json]: Alright, I will create an issue for this internally. For now it is best te enable insight I think, otherwise users would have to re-pair to enable it.

Comment (RobinBol) [reply app.json]: FYI, this issue is fixed in the next homey-client firmware version.

File: .homeyplugins.json (hunk 2 above)
Comment (TedTolboom) [reply .homeyplugins.json]: Fixed in commit 4a59acd

File: app.json (hunk 3 above)
Comment (TedTolboom) [reply app.json]: Resolved with commit 856f43f

File: app.json (hunk 4 above)
Comment (TedTolboom) [reply app.json]: Do I need to remove it?

File: app.json (hunk 5 above)
Comment (TedTolboom) [reply app.json]: Resolved with b25ada2

File: app.json (hunk 6 above)
Comment (TedTolboom) [reply app.json]: I've re-used here the default 'autoOff' setting...

File: app.json (hunk 7 above)
Comment (TedTolboom) [reply app.json]: See comment for parameter 11 ;-P

File: app.json (hunk 8 above)
Comment (TedTolboom) [reply app.json]: Z-wave refers to the controller... replaced by Homey (Z-wave)
Resolved in commit 4fa0082

File: app.json (hunk 9 above)
Comment (TedTolboom) [reply app.json]: Fixed in commit c5409a7

File: app.json (hunk 10 above)
Comment (TedTolboom) [reply app.json]: Fixed in 660392

File: drivers/ZMNHQD/assets/icon.svg (hunk 11 above)
Comment (TedTolboom) [reply drivers/ZMNHQD/assets/icon.svg]: Works fine here... requested an Ivo icon just now.

File: drivers/ZMNHQD/device.js (hunk 12 above)
Comment (TedTolboom) [reply drivers/ZMNHQD/device.js]: removed link, kept placeholder.
commit 3fb8902

File: drivers/ZMNHQD/device.js (hunk 13 above)
Comment (TedTolboom) [reply drivers/ZMNHQD/device.js]: removed in commit 3fb8902

File: drivers/ZMNHQD/device.js (hunk 15 above)
Comment (TedTolboom) [reply drivers/ZMNHQD/device.js]: Fixed in commit 3fb8902

File: drivers/ZMNHQD/device.js (hunk 18 above)
Comment (TedTolboom) [reply drivers/ZMNHQD/device.js]: will remove

File: app.json (hunk 23 above)
Comment (TedTolboom) [reply app.json]: in ZMNHQD/device.js

```
// determine dim duration
let dimDuration = this.getSetting('dimDuration') * 1000;
```

File: drivers/ZMNHQD/device.js (hunk 14 above)
Comment (TedTolboom) [reply drivers/ZMNHQD/device.js]: I've installed the eslint config as advised, but this also installs a sh*t load of node_modules... which are not part of the current repository; do you want me to commit them?

File: lib/QubinoDriver.js (hunk 21 above)
Comment (TedTolboom) [reply lib/QubinoDriver.js]: Reason to add it to the QubinoDriver is that there will be a second LUXY device that will be released on short term.. requiring the same _registerFlowCardTriggerDevice()

File: lib/QubinoDriver.js (hunk 22 above)
Comment (TedTolboom) [reply lib/QubinoDriver.js]: Reason to add it to the QubinoDriver is that there will be a second LUXY device that will be released on short term.. requiring the same _registerFlowCardTriggerDevice()

File: app.json (hunk 23 above)
Comment (TedTolboom) [reply app.json]: So it calls for the stored setting, and if there is a duration added by the flow card, it is being overwritten... should almost be a default behavior ;-P

Comment (RobinBol) [reply app.json]: Ah I see, thanks for clarifying!

**Comment by RobinBol** [comment]
Hi @TedTolboom, I made some changes in master which I merged into your feature branch so it will remain mergeable. Your changes seem good to me, however, I think you forgot to compose (some differences between app.json and the compose files). Could you fix this and let me know what you think is the status of this PR (mergeable, or are you still working on it)?

On a side note, we decided to disable S0 security for the drivers in this app for the time being. We got some feedback that this was causing problems during inclusion (as you mentioned as well).

**Comment by TedTolboom** [comment]
Hi Robin,

I've finalized the device driver, updated the icon (minor changes by Ivo) and images.
Can you do a final review?

Thanks, Ted

File: app.json [hunk 24]
```diff
@@ -10397,14 +10396,14 @@
...
-            "nl": "(Her-)activeer volledig wit (een keer drukken op (1)). \n\nWanneer volledig wit geactiveerd is, druk op rechts (2) en houdt ingedrukt tussen 4 en 6 seconden.\n\nNa 6 seconden zal LUXY Smart Light (ZMNHQD) groen gaan knipperen, 1 seconde aan, 0.5 seconde uit.\n\n Wanneer toevoegen succesvol afgerond is (na 10 seconden), stopt het knipperen en wordt LED volledig groen."
+            "en": "(Re-)activate full white (one press on (1)). \nWhen full white is enabled, press right (2) and hold it in between 4 and 6 seconds. \nAfter 6 seconds, the LUXY Smart Light (ZMNHQD) starts flashing green 1 second on, 0.5 second off.",
+            "nl": "(Her-)activeer volledig wit (een keer drukken op (1)). \nWanneer volledig wit geactiveerd is, druk op rechts (2) en houdt ingedrukt tussen 4 en 6 seconden.\nNa 6 seconden zal LUXY Smart Light (ZMNHQD) groen gaan knipperen, 1 seconde aan, 0.5 seconde uit."
```
Comment (RobinBol) [inline app.json]: Why did you remove one of the newline characters? I don't think this improves readability tbh, maybe best to stick to the standard in this app?

```
1. Step one... \n\n
2. Step two... \n\n
...
```
```

File: app.json [hunk 25]
```diff
@@ -10572,29 +10571,29 @@
...
+                "en": "Capacitive input & Homey (Z-wave)",
+                "nl": "Capacitieve bedieningsknoppen & Homey (Z-wave)"
               }
             }
           ],
           "hint": {
-            "en": "This parameter (3) determines how the alarm (buzzer) can be disabled.",
-            "nl": "Deze parameter (3) specificeert hoe het alarm (geluid) uitgeschakeld kan worden."
+            "en": "This parameter (3) determines how the buzzer can be disabled.",
+            "nl": "Deze parameter (3) specificeert hoe het geluid uitgeschakeld kan worden."
```
Comment (RobinBol) [inline app.json]: Here you are referring to 'geluid', everywhere else it is 'buzzer', best to stay consistent.

File: app.json [hunk 26]
```diff
@@ -10609,8 +10608,8 @@
             "nl": "Bewegingssensor uitschakeltijd"
           },
           "hint": {
-            "en": "This parameter (5) determines the time after which the output triggered by motion sensor is turned off. \n32535 – never turns off",
-            "nl": "Deze parameter (5) bepaalt de tijdsduur waarna de uitgang die ingeschakeld is doot de Bewegingssensor uitgeschakeld wordt. \n32535 – schakelt nooit uit"
+            "en": "This parameter (5) determines the time after which the output triggered by motion sensor is turned off. Setting the value to 32535 will result in the output never being turned off.",
+            "nl": "Deze parameter (5) bepaalt de tijdsduur waarna de uitgang die ingeschakeld is doot de Bewegingssensor uitgeschakeld wordt. Het instellen van de waarde op 32535 zorgt dat de uitgang nooit uitgeschakeld wordt."
```
Comment (RobinBol) [inline app.json]: door*
bewegingssensor*

File: app.json [hunk 27]
```diff
@@ -10621,6 +10620,39 @@
...
+              "label": {
+                "en": "Disabled (the buzzer is not active)",
+                "nl": "Uitgeschakeld (de buzzer is inactief)"
+              }
+            },
+            {
+              "id": "1",
+              "label": {
+                "en": "Enabled (the buzzer is active)",
+                "nl": "ingeschakeld (de buzzer is actief)"
```
Comment (RobinBol) [inline app.json]: Ingeschakeld*

Comment (RobinBol) [inline app.json]: The postfix ('is active') is not really necessary, and often not readable in the UI.

File: app.json [hunk 28]
```diff
@@ -10621,6 +10620,39 @@
...
+        {
+          "id": "buzzerFunction",
+          "type": "dropdown",
+          "zwave": {
+            "index": 3,
+            "size": 1
+          },
+          "label": {
+            "en": "Buzzer function",
+            "nl": "Buzzer functie"
```
Comment (RobinBol) [inline app.json]: `Buzzer function` is not really an effective label, besides a dropdown is not the best type for this setting. Why not make it a checkbox with label "Buzzer ingeschakeld"?

File: app.json [hunk 29]
```diff
@@ -10621,6 +10620,39 @@
...
+              "id": "1",
+              "label": {
+                "en": "Enabled (the buzzer is active)",
+                "nl": "ingeschakeld (de buzzer is actief)"
+              }
+            }
+          ],
+          "hint": {
+            "en": "This parameter (3) affects buzzer function just in case when the device receive the NOTIFICATION_REPORT.",
+            "nl": "Deze parameter (3) specificeert de buzzer functie in het geval dat het apparaat een NOTIFICATION_REPORT ontvangt."
```
Comment (RobinBol) [inline app.json]: receives*

This hint is not really clear.

File: drivers/ZMNHQD/device.js [hunk 30]
```diff
@@ -46,32 +46,45 @@ class ZMNHQD extends QubinoDimDevice {
       },
     });
 
+    // Initialize Buzzer state
+    let BUZZER_STATE = this.getCapabilityValue(CAPABILITIES.ONOFF_BUZZER);
```
Comment (RobinBol) [inline drivers/ZMNHQD/device.js]: CAPITAL_CASE is a convention for constants, best to make it camelCase as it is a variable here.

**Comment by RobinBol** [comment]
We are nearly there Ted, just some final comments

File: app.json (hunk 27 above)
Comment (TedTolboom) [reply app.json]: Changed to "en": "Turn off buzzer", /  "nl": "Zet buzzer uit"

File: drivers/ZMNHQD/device.js (hunk 30 above)
Comment (TedTolboom) [reply drivers/ZMNHQD/device.js]: changed...

File: app.json (hunk 29 above)
Comment (TedTolboom) [reply app.json]: updated

File: app.json (hunk 28 above)
Comment (TedTolboom) [reply app.json]: changed to checkbox and with updated label

File: app.json (hunk 27 above)
Comment (TedTolboom) [reply app.json]: obsolete with changing to checkbox

Comment (TedTolboom) [reply app.json]: obsolete with changing to checkbox

File: app.json (hunk 25 above)
Comment (TedTolboom) [reply app.json]: updated

File: app.json (hunk 24 above)
Comment (TedTolboom) [reply app.json]: Updated in latest commit

File: app.json [hunk 31]
```diff
@@ -414,8 +414,8 @@
       {
         "id": "turnOnBuzzer_ZMNHQD",
         "title": {
-          "en": "Activate alarm buzzer",
-          "nl": "Schakel alarm buzzer in"
+          "en": "Activate buzzer",
+          "nl": "Schakel buzzer in"
```
[the remark this answers is not in the document]
Comment (TedTolboom) [reply app.json]: checked all and made consistent

**Comment by TedTolboom** [comment]
Resolved all issues in commit 0f00552

File: lib/QubinoDriver.js (hunk 21 above)
Comment (RobinBol) [reply lib/QubinoDriver.js]: Alright, in that case! Are you sure you want to leave this comment in this public repo?

**Comment by RobinBol** [comment]
@TedTolboom thanks, I found some more minor things and was so free to fix them myself (to speed things up a bit). Do you agree with the changes? If so we can merge this PR and publish it to test.
